### ISSUE TITLE
When reporting health for a node use the {node.name}{node.id} as the key

### ISSUE THREAD

**Issue by andreidan** [elastic/elasticsearch#89664]
The health API currently reports the cluster formation view of different nodes
keyed by the node.id
e.g.
```
"cluster_formation": {
          "0oN5vgNOTYSD2e2NLWcMpA": "master not discovered yet: have discovered [{node_t2}{0oN5vgNOTYSD2e2NLWcMpA}{FG_hf0jaTuihP9P-YwQY7g}{node_t2}{127.0.0.1}{127.0.0.1:13422}{d}]; discovery will continue using [127.0.0.1:13421] from hosts providers and [{node_t0}{V82Zl-CUS_ah7O0-JIBhfQ}{McMwzFMoRui2n4y2rUif6g}{node_t0}{127.0.0.1}{127.0.0.1:13421}{m}] from last-known cluster state; node term 1, last-accepted version 7 in term 1"
}
```

This proposes we change the key to be a combination of node.name and node.id for ease of use:
i.e.
```
"cluster_formation": {
          "{node_t2}{0oN5vgNOTYSD2e2NLWcMpA}": "master not discovered yet: have discovered [{node_t2}{0oN5vgNOTYSD2e2NLWcMpA}{FG_hf0jaTuihP9P-YwQY7g}{node_t2}{127.0.0.1}{127.0.0.1:13422}{d}]; discovery will continue using [127.0.0.1:13421] from hosts providers and [{node_t0}{V82Zl-CUS_ah7O0-JIBhfQ}{McMwzFMoRui2n4y2rUif6g}{node_t0}{127.0.0.1}{127.0.0.1:13421}{m}] from last-known cluster state; node term 1, last-accepted version 7 in term 1"
}
```

**Comment by dakrone** [comment]
we have precedence internally to use `[]` as delimiters, what about doing `[<node-name>/<node-id>]` for the key? This is similar to what we do for `Index` instances with `[<index-name>/<index-uuid>]` in our logs.

**Comment by andreidan** [comment]
@dakrone the `DiscoveryNode` description seems to rely on `{ }` https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java#L566

I think we should follow that pattern - also the key ( `{node_t2}{0oN5vgNOTYSD2e2NLWcMpA}` ) would be copy-pastable 
```
"{node_t2}{0oN5vgNOTYSD2e2NLWcMpA}": "master not discovered yet: have discovered [{node_t2}{0oN5vgNOTYSD2e2NLWcMpA}{FG_hf0jaTuihP9P-YwQY7g}{node_t2}{127.0.0.1}{127.0.0.1:13422}{d}]; discovery will continue using [127.0.0.1:13421] from hosts providers and [{node_t0}{V82Zl-CUS_ah7O0-JIBhfQ}{McMwzFMoRui2n4y2rUif6g}{node_t0}{127.0.0.1}{127.0.0.1:13421}{m}] from last-known cluster state; node term 1, last-accepted version 7 in term 1"
}
```

**Comment by andreidan** [comment]
Closed by #89982 (we decided to return both the node id and node name under `cluster_formation`)
